server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -711,10 +716,12 @@ private static <T, QueryType extends Query<T>> DataSource toInlineDataSource(
           subqueryStatsProvider.incrementQueriesExceedingByteLimit();
           throw ResourceLimitExceededException.withMessage(byteLimitExceededMessage(memoryLimit));
         }
+        AtomicReference<Sequence<T>> fallbackSequence = new AtomicReference<>(null);
```

Comment:
note: I feel like you are trying to pre-communicate a return value;  this seem like some attached detail which should be recovered either via return value or by hauled in an exception.

what if `materializeResultsAsFrames` doesn't set `fallbackSequence`? like when it handles [UnsupportedColumnTypeException](https://github.com/apache/druid/pull/16800/files#diff-073196043cf52e8c14ca92d8df79bc9e17b20e6df3fb60ec0a97f78b5d896595R829-R833) right now - will it be okay to pass `null` to `materializeResultsAsArray` ?